src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonDocumentTests.cs

```diff
@@ -3437,6 +3453,16 @@ public static void ParseValue_AllowMultipleValues_TrailingContent()
             JsonTestHelper.AssertThrows<JsonException>(ref reader, (ref Utf8JsonReader reader) => reader.Read());
         }
 
+        [Theory]
+        [InlineData("""{ "foo" : [1], "test": false, "bar" : { "nested": 3 } }""", 3)]
+        [InlineData("""{ "foo" : [1,2,3,4] }""", 1)]
+        [InlineData("""{}""", 0)]
+        [InlineData("""{ "foo" : {"nested:" : {"nested": 1, "bla": [1, 2, {"bla": 3}] } }, "test": true, "foo2" : {"nested:" : {"nested": 1, "bla": [1, 2, {"bla": 3}] } }}""", 3)]
+        public static void TestGetPropertyCount(string json, int expectedCount) {
+            JsonElement element = JsonSerializer.Deserialize<JsonElement>(json);
+            Assert.Equal(expectedCount, element.GetPropertyCount());
+        }
+
```

Comment:
I recommend adding a test that loads an arbitrarily large/complex document (which we probably already have one), and walks deeply showing that GetPropertyCount() and EnumerateObject().Count() are the same at every object node.